### PR TITLE
BC-3567 - MVP Visibility of Task Cards

### PR DESCRIPTION
**Pull request by MartinSchuhmacher**
# Short Description
Implement visibility MVP for task cards as discussed with UX.

## Links to Ticket and related Pull-Requests
BC-3567
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3997

## Changes
- add visibility into task card view
- make select component disabled for now
- set default visibility as "visible" for created task cards

## Screenshots of UI changes

View with edit permission
image

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/TaskCard.page.vue [hunk 1]
```diff
@@ -129,6 +138,17 @@ export default defineComponent({
 
 		const course = ref("");
 		const courses = ref<object[]>([]);
+		const isVisible: Ref<boolean> = ref(true);
```
Comment (christian-darsow) [inline src/pages/TaskCard.page.vue:141]: Is this explicit type setting needed here? or can we simply use inferred type?

File: src/pages/TaskCard.page.vue [hunk 2]
```diff
@@ -182,6 +202,7 @@ export default defineComponent({
 						title: taskCardData.courseName || "",
 					},
 				];
+				isVisible.value = !taskCardData.task.status.isDraft;
```
Comment (christian-darsow) [inline src/pages/TaskCard.page.vue:205]: Just as I wonder: can this lead to invisible beta task for now or will isDraft always be false?

File: src/pages/TaskCard.page.vue (hunk 1 above)
Comment (MartinSchuhmacher) [reply src/pages/TaskCard.page.vue:141]: I just typed it explicit to make the type of value clear here, no specific needing, just own preference :)

File: src/pages/TaskCard.page.vue (hunk 2 above)
Comment (MartinSchuhmacher) [reply src/pages/TaskCard.page.vue:205]: together with this change in the BE the beta task will now be always visible for now https://github.com/hpi-schul-cloud/schulcloud-server/pull/3997/files#diff-86ca587a2205b44debc99574bb96bd361ffa567409783f493a3e046e0f2692f2R85

as documented in https://ticketsystem.dbildungscloud.de/browse/BC-2807

**Review by odalys-dataport — DISMISSED** [review]

**Review by christian-darsow — DISMISSED** [review]

**Pull request merged**